### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by jgong5** [libxsmm/libxsmm#819]
Does libxsmm support Windows?

**Comment by alheinecke** [comment]
yes, but we don't validate / CI on Windows. Do you run into issues? There might be an issue with some callee-save registers. I'm tracking this as a low-prio issue right now.

**Comment by jgong5** [comment]
> yes, but we don't validate / CI on Windows. Do you run into issues? There might be an issue with some callee-save registers. I'm tracking this as a low-prio issue right now.

We are enabling IPEX on Windows but we haven't enabled the tpp/libxsmm related features in IPEX right now. There are some compilation issues we need to look into. Just wanted to confirm its support here. May I know if you will add this into you validation? Also, may I know what are impacted with the callee-save register issue?

**Comment by jgong5** [comment]
cc @xuhancn

**Comment by hfp** [comment]
> We are enabling IPEX on Windows but we haven't enabled the tpp/libxsmm related features in IPEX right now. There are some compilation issues we need to look into. Just wanted to confirm its support here.

> May I know if you will add this into you validation?

It is possible of course, but low priority on our side. In principle, I prepared our Makefile based build system compatible with Windows (spaces in path, etc). I am also keeping VS projects up-to-date at least wrt latest VS. The latter can be automated too, and there are a few Open Source CI-provider operating on Windows. Beside of the latter, we can likely get cloud support to do so on our own.

> Also, may I know what are impacted with the callee-save register issue?

I did this work a very long time ago, and our regmapping was not updated since then even when we changed our calling convention/interface for kernels. Assume it may work by luck.
